internal/eventstore/eventstore.go

```diff
@@ -77,7 +83,23 @@ func (es *Eventstore) Push(ctx context.Context, cmds ...Command) ([]Event, error
 		ctx, cancel = context.WithTimeout(ctx, es.PushTimeout)
 		defer cancel()
 	}
-	events, err := es.pusher.Push(ctx, cmds...)
+	var (
+		events []Event
+		err    error
+	)
+
+	// Retry when there is a collision of the sequence as part of the primary key.
+	// "duplicate key value violates unique constraint \"events2_pkey\" (SQLSTATE 23505)"
+	// https://github.com/zitadel/zitadel/issues/7202
+retry:
+	for i := 0; i <= es.maxRetries; i++ {
+		events, err = es.pusher.Push(ctx, cmds...)
+		var pgErr *pgconn.PgError
+		if !errors.As(err, &pgErr) || pgErr.ConstraintName != "events2_pkey" || pgErr.SQLState() != "23505" {
+			break retry
+		}
+		logging.WithError(err).Info("eventstore push retry")
+	}
```

Comment:
As discussed I would first try to set the isolation level of the transaction [here](https://github.com/zitadel/zitadel/blob/a88662bb3a50e48dae06b41705fd9ee342022a1a/internal/eventstore/v3/push.go#L22) to serialisable and set the max retry to the configured one

Reply:
As discussed, using `LevelSerializable` instead of retry brings back the bug.